### DISCUSSION TITLE
Will ruff check support inputting code text for static analysis in future versions?

### DISCUSSION THREAD

**Discussion by helloMr996** [astral-sh/ruff#14595] [Ideas]
Processing individual code texts from the LLM Python dataset with ruff for code quality filtering, but currently ruff only supports detection of Python files or directories. Will it be possible to support code text input detection in the future?

**Comment by MichaReiser** [comment]
> Will it be possible to support code text input detection in the future?

I don't think I understand your suggestion. Would you mind telling me more about what a *code text*, could you share a concrete example?

**Reply by helloMr996** [reply]
For example, in the case of the "the-stack-v2" Python dataset, each document within the dataset can be used as input for static analysis, rather than analyzing through code files.

**Reply by MichaReiser** [reply]
I don't have access to the-stack-v2 python data set and I don't know how it is structured. 

Can you tell me more about your use case? What are you trying to accomplish and why?

**Reply by helloMr996** [reply]
The the-stack-v2 Python dataset is a JSONL (JSON Lines) format file, in which there is a text field whose value is a string of code text.              I want to use Ruff to improve the quality of the code dataset and then use it for training large code models.

**Reply by MichaReiser** [reply]
Thanks. That was helpful.

I don't think we'll add support for such specific data formats to ruff itself. It seems too niche. I suggest you write a small script to extract the files yourself and then pipe the files to Ruff itself using `ruff check --stdin-filename="name" < content.txt`. Or you use the published JS API or you could consider contributing a [Python API](https://github.com/astral-sh/ruff/issues/659)

**Reply by helloMr996** [reply]
Thank you very much!
